main/src/ca/uwaterloo/flix/language/ast/SimplifiedAst.scala

```diff
@@ -264,6 +264,22 @@ object SimplifiedAst {
 
   }
 
+  case class LabeledExpression(label: Int, exp : Expression) extends Expression {
+    override def tpe: Type = exp.tpe
+
+    override def loc: SourceLocation = exp.loc
+
+    override def equals(obj: Any): Boolean =
+      obj match {
+        case that: LabeledExpression => this.label == that.label
+        case _ => false
+      }
+
+    override def hashCode(): Int = label.hashCode()
+  }
+
+  implicit def expToLabeled(exp: Expression): LabeledExpression = exp.asInstanceOf[LabeledExpression]
```

Comment:
This kind of implicit coercion is the worst kind of abuse :)

But also, I thought that we agreed this kind of Wrapper expression does not work anyway?